### PR TITLE
3.2a

### PR DESCRIPTION
**Pull request by theQuazz**
- remove old school duel
- add -respawn-system
- fix docker setup for those OSX users (me)

### CODE REVIEW

File: wurst/systems/modes/Respawn.wurst [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import GameConfig
+import TimerUtils
+import TimerDialog
+
+@configurable constant GRACE_PERIOD_REVIVE_DELAY = 10.0
+
+function isGracePeriodActive() returns boolean
+    return udg_NOOB_TIME
+
+function quadraticReviveDelay(real x) returns real
```
Comment (jonathanmclaus) [inline wurst/systems/modes/Respawn.wurst]: Change `x` to `seconds` or `num_seconds`. I'd prefer the latter but it's whatever.

Comment (theQuazz) [reply wurst/systems/modes/Respawn.wurst]: We use `camelCase` not `snake_case` so I'll switch it to `numSeconds` and `numMinutes`

File: wurst/systems/modes/Respawn.wurst [hunk 2]
```diff
@@ -1,16 +1,19 @@
...
 import TimerUtils
 import TimerDialog
+import ChatCommands
+import LinkedList
 
 @configurable constant GRACE_PERIOD_REVIVE_DELAY = 10.0
 
 function isGracePeriodActive() returns boolean
-    return udg_NOOB_TIME
+    return gameConfig.isGracePeriodEnabled() and udg_NOOB_TIME
```
Comment (jonathanmclaus) [inline wurst/systems/modes/Respawn.wurst]: Are these not redundant?

Comment (theQuazz) [reply wurst/systems/modes/Respawn.wurst]: No, one refers to the modes of the game and the other refers to whether noob time is still active

Comment (jonathanmclaus) [reply wurst/systems/modes/Respawn.wurst]: Yes, but grace period is not enabled, isn't `udg_NOOB_TIME` always false?

Comment (theQuazz) [reply wurst/systems/modes/Respawn.wurst]: Actually `udg_NOOB_TIME` is default `true` so hence the check

**Comment by theQuazz** [comment]
@jonathanmclaus I've updated the PR to remove `udg_NOOB_TIME`

File: wurst/systems/modes/Respawn.wurst [hunk 3]
```diff
@@ -80,6 +82,25 @@ init
...
                     delayTimer.doAfter(delay) ->
-                        revive(dyingUnit, owner)
                         timerDialog.display(false)
                         timerDialog.destr()
+
+                        let units = ENUM_GROUP
+                            ..enumUnitsOfForce(owner.getAllies())
+                            .toLinkedList()
+                            ..sortWith((a, b) -> begin
+                                let aType = a.isType(UNIT_TYPE_STRUCTURE) ? 1 : 0
```
Comment (jonathanmclaus) [inline wurst/systems/modes/Respawn.wurst]: Could we cast a boolean into an integer here? If so, I think that's more readable.

File: wurst/systems/modes/Respawn.wurst [hunk 4]
```diff
@@ -80,6 +82,25 @@ init
...
+                                let bType = a.isType(UNIT_TYPE_STRUCTURE) ? 1 : 0
+                                return aType == bType ?
```
Comment (jonathanmclaus) [inline wurst/systems/modes/Respawn.wurst]: I'm not sure what your personal taste is, but I think it is generally recommended (and I prefer) to start both subsequent lines with the operator.

    return aType == bType
        ? a.getLevel() - b.getLevel()
        : aType - bType

Comment (theQuazz) [reply wurst/systems/modes/Respawn.wurst]: I'm ok with that

**Pull request merged**